navigation-animation/src/main/java/com/google/accompanist/navigation/animation/AnimatedNavHost.kt

```diff
@@ -0,0 +1,270 @@
+        exitTransition,
+    builder: NavGraphBuilder.() -> Unit
+) {
+    AnimatedNavHost(
+        navController,
+        remember(route, startDestination, builder) {
+            navController.createGraph(startDestination, route, builder)
+        },
+        modifier,
+        enterTransition,
+        exitTransition,
+        popEnterTransition,
+        popExitTransition
+    )
+}
+
+/**
+ * Provides in place in the Compose hierarchy for self contained navigation to occur.
+ *
+ * Once this is called, any Composable within the given [NavGraphBuilder] can be navigated to from
+ * the provided [navController].
+ *
+ * @param navController the navController for this host
+ * @param graph the graph for this host
+ * @param modifier The modifier to be applied to the layout.
+ * @param enterTransition callback to define enter transitions for destination in this host
+ * @param exitTransition callback to define exit transitions for destination in this host
+ * @param popEnterTransition callback to define popEnter transitions for destination in this host
+ * @param popExitTransition callback to define popExit transitions for destination in this host
+ */
+@ExperimentalAnimationApi
+@Composable
+public fun AnimatedNavHost(
+    navController: NavHostController,
+    graph: NavGraph,
+    modifier: Modifier = Modifier,
+    enterTransition: (initial: NavBackStackEntry, target: NavBackStackEntry) -> EnterTransition =
+        { _, _ -> fadeIn(animationSpec = tween(2000)) },
+    exitTransition: (initial: NavBackStackEntry, target: NavBackStackEntry) -> ExitTransition =
+        { _, _ -> fadeOut(animationSpec = tween(2000)) },
+    popEnterTransition: (initial: NavBackStackEntry, target: NavBackStackEntry) -> EnterTransition =
+        enterTransition,
+    popExitTransition: (initial: NavBackStackEntry, target: NavBackStackEntry) -> ExitTransition =
+        exitTransition,
+) {
+
+    enterTransitions[graph.route] = enterTransition
+    exitTransitions[graph.route] = exitTransition
+    popEnterTransitions[graph.route] = popEnterTransition
+    popExitTransitions[graph.route] = popExitTransition
+
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val viewModelStoreOwner = checkNotNull(LocalViewModelStoreOwner.current) {
+        "NavHost requires a ViewModelStoreOwner to be provided via LocalViewModelStoreOwner"
+    }
+    val onBackPressedDispatcherOwner = LocalOnBackPressedDispatcherOwner.current
+    val onBackPressedDispatcher = onBackPressedDispatcherOwner?.onBackPressedDispatcher
+
+    // on successful recompose we setup the navController with proper inputs
+    // after the first time, this will only happen again if one of the inputs changes
+    navController.setLifecycleOwner(lifecycleOwner)
+    navController.setViewModelStore(viewModelStoreOwner.viewModelStore)
+    if (onBackPressedDispatcher != null) {
+        navController.setOnBackPressedDispatcher(onBackPressedDispatcher)
+    }
+
+    navController.graph = graph
```

Comment:
This should be done in an effect? It's updating global state in a composition which might not be 'committed'

Reply:
This is also expected at this point.